lib/protein_utils.rb

```
@@ -0,0 +1,52 @@
+class ProteinUtils
+
+  def self.translate_to_polypeptide(array)
+    polypeptides = array.map{ |key| codon_to_polypeptide(key) }
+  end
+
+  def self.proteins(array)
+    array.include?('') ? reject_terminating_codons(array) : array
+  end
+
+  def self.reject_terminating_codons(array)
+    terminating_index = array.find_index('')
+    array.delete_if.with_index{ |codon, index| index >= terminating_index }
+  end
+
+  def self.codon_to_polypeptide(codon)
+    polypeptide = case codon.upcase
+                  when 'AUG' then 'Methionine'
+                  when 'UUU', 'UUC' then 'Phenylalanine'
+                  when 'UUA', 'UUG' then 'Leucine'
+                  when 'UCU', 'UCC', 'UCA', 'UCG' then 'Serine'
+                  when 'UAU', 'UAC' then 'Tyrosine'
+                  when 'UGU', 'UGC' then 'Cysteine'
+                  when 'UGG' then 'Tryptophan'
+                  when 'UAA', 'UAG', 'UGA' then ''
+                  else 'Oops! Something went wrong. Try again.'
```

You might want to `raise` here instead of just returning the string `Oops...`

---

You can do something like:

```ruby 
expect do 
  described_class.codon_to_polypeptide('FOOBAR')
end.to raise_error
```